core/templates/dev/head/pages/exploration_player/HintsTooltipService.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for displaying the hints tooltip to the learner.
+ */
+
+oppia.factory('HintsTooltipService', [
+  '$uibModal', '$uibModalStack', '$timeout', 'UrlInterpolationService',
+  function($uibModal, $uibModalStack, $timeout, UrlInterpolationService) {
```

Comment:
Just as a reminder: after the tooltip is shown once (and then dismissed or timed out), or a hint is ever opened, the tooltip should then never be viewable again for that entire exploration. (Thought I'd mention this because the logic here seems a bit complicated -- I thought it'd basically be showTooltip() and then that sets a timeout for closing the tooltip but also sets a flag saying never open the tooltip again. And then I don't think you really need to expose any other stuff...)